workflow/controller/controller.go

```diff
@@ -353,12 +353,7 @@ func (wfc *WorkflowController) notifySemaphoreConfigUpdate(cm *apiv1.ConfigMap)
 			log.Warnf("received object from indexer %s is not an unstructured", indexes.SemaphoreConfigIndexName)
 			continue
 		}
-		wf, err := util.FromUnstructured(un)
```

Comment:
Fix (a): `util.FromUnstructured` is EXPENSIVE. We should avoid it when we can and just use the unstructured.